google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc

```diff
@@ -690,24 +690,26 @@ void SetNativeIamPolicy(google::cloud::bigtable::InstanceAdmin instance_admin,
   (std::move(instance_admin), argv.at(0), argv.at(1), argv.at(2));
 }
 
-void TestIamPermissions(std::vector<std::string> argv) {
+void TestIamPermissions(std::vector<std::string> const& argv) {
   if (argv.size() < 3) {
     throw Usage{
         "test-iam-permissions <project-id> <resource-id>"
         " <permission> [permission ...]"};
   }
-  std::string resource = argv[1];
+  auto it = argv.cbegin();
+  auto const project_id = *it++;
+  auto const resource = *it++;
+  std::vector<std::string> const permissions(it, argv.cend());
```

Comment:
I assume that we don't care too much about copying in the cases like this.  Otherwise, consider something that starts with:
```
std::rotate(argv.begin(), argv.begin() + 2, argv.end());
```